docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,485 @@
+# Danceable Lyrics
+In this guide, we use the [Alteryx Python SDK](https://pypi.org/project/ayx-python-sdk/) and [Alteryx Plugin CLI](https://pypi.org/project/ayx-plugin-cli/) to create tool that connects directly to a set of data files and determines the most "danceable" songs containing a set of lyrics. We will be utilizing [Polars](https://www.pola.rs/) for fast data processing.
+
+# Download Input Data
+Download and extract the required dataset files. You will reference them later on in this guide. We provide to sets of this data -- the full and a truncated, smaller version.
+
+* The full version expands to ~11 GiB of data and [can be downloaded here](https://drop.alteryx.com/public/file/SAlShV5VqkOI5_Jyx1me3w/Danceable_Lyrics_demo_data-Inspire-2023.7z).
+* A smaller truncated version that expands to ~5.7 GiB of data [can be downloaded here](https://drop.alteryx.com/public/file/yHv77vLzKkCXHj8-OWVg0Q/Danceable_Lyrics_demo_data-Inspire-(Truncated)-2023.7z).
+
+See the `README.md` within the respective archives for source information.
+
+# Basic Plugin Setup
+Before proceeding, please ensure you have the following basic setup:
+
+1. [Create a Workspace](create-a-workspace.md) for your plugin to live within.
+2. [Add a new plugin](create-a-plugin.md).
+    1. When prompted choose **Input** as the type.
+    2. To match what is found in this guide, choose **DanceableLyrics** as the plugin name.
+
+After running the above setup procedures, you will see a file named `danceable_lyrics.py` under `./backend/ayx_plugins` with a generated boilerplate. When you open the file, should see something like this:
+
+```python
+class DanceableLyrics(PluginV2):
+    """Concrete implementation of an AyxPlugin."""
+
+    def __init__(self, provider: AMPProviderV2) -> None:
+        self.provider = provider
+        # truncated code
+
+    def on_incoming_connection_complete(self, anchor: namedtuple) -> None:
+        # truncated code
+
+    def on_record_batch(self, batch: "Table", anchor: namedtuple) -> None:
+        # truncated code
+
+    def on_complete(self) -> None:
+        import pandas as pd
+        import pyarrow as pa
+
+        df = pd.DataFrame(
+            {
+                "x": [1, 2, 3],
+                "y": ["hello", "world", "from ayx_python_sdk!"],
+                "z": [self.config_value, self.config_value, self.config_value],
+            }
+        )
+
+        packet = pa.Table.from_pandas(df)
+
+        self.provider.write_to_anchor("Output", packet)
+        self.provider.io.info("DanceableLyrics tool done.")
+```
+
+# Write a Plugin!
+In this guide, we will be utilzing multiple data files, reading each directly illustrating a "connector" like plugin. That is, to connect directly to local or remote data sources. While we will be reading CSV input files, these could be in any format including those not directly supported by Designer!
```

Comment:
TW review:

In this guide, we use multiple data files. We read each directly to illustrate a connector-like plugin. That is, to connect directly to local or remote data sources. While here, we read CSV input files, these could be in any format, including formats that Alteryx Designer does not directly support.